lib/receive.dart

```diff
@@ -202,78 +158,157 @@ class MaybeInputsSeen extends FfiV2MaybeInputsSeen {
   }
 }
 
-class OutputsUnknown extends FfiV2OutputsUnknown {
+class OutputsUnknown extends FfiOutputsUnknown {
   OutputsUnknown._({required super.field0});
 
   /// Find which outputs belong to the receiver
   @override
-  Future<ProvisionalProposal> identifyReceiverOutputs(
+  Future<WantsOutputs> identifyReceiverOutputs(
       {required FutureOr<bool> Function(Uint8List p1) isReceiverOutput,
       hint}) async {
     try {
       final res = await super
           .identifyReceiverOutputs(isReceiverOutput: isReceiverOutput);
-      return ProvisionalProposal._(field0: res.field0);
+      return WantsOutputs._(field0: res.field0);
     } on error.PayjoinError catch (e) {
       throw mapPayjoinError(e);
     }
   }
 }
 
-class ProvisionalProposal extends FfiV2ProvisionalProposal {
-  ProvisionalProposal._({required super.field0});
+class WantsOutputs extends FfiWantsOutputs {
+  WantsOutputs._({required super.field0});
+
+  @override
+  Future<bool> isOutputSubstitutionDisabled({hint}) {
+    try {
+      return super.isOutputSubstitutionDisabled();
+    } on error.PayjoinError catch (e) {
+      throw mapPayjoinError(e);
+    }
+  }
 
   @override
-  Future<void> trySubstituteReceiverOutput(
-      {required FutureOr<Uint8List> Function() generateScript}) {
+  Future<WantsOutputs> replaceReceiverOutputs(
+      {required List<TxOut> replacementOutputs,
+      required FfiScript drainScript}) async {
```

Comment:
I think it is better to use the Script type here, since FfiScript is an internal type and the user will need to use the Script class anyways to create a new instance for FfiScript that it extends.